resize hash table before building

englefly opened the pull request:
# Proposed changes
Initialize hash table size by the tuple number instead of fixed number 1024 to reduce BuildTableExpanseTime.
After initialize table size, the total build time decreased by 8.9% on tpch 10G, 
`select count(*) from lineitem join orders on l_orderkey = o_orderkey`

Issue Number: close #xxx

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

zbtzbtzbt commented:
what is the hashtable build time? before and after your pr.

zbtzbtzbt commented on be/src/vec/common/hash_table/hash_table.h:
```diff
@@ -612,6 +555,62 @@ class HashTable : private boost::noncopyable,
 
+        /// Increase the size of the buffer.
+    void resize(size_t for_num_elems = 0, size_t for_buf_size = 0) {
+        SCOPED_RAW_TIMER(&_resize_timer_ns);
```
why move this function? it seems no changes

englefly replied:
HashTable::resize  moved from protected to public, so that it could be invoked by ProcessHashTableBuild in vhash_join_node.cpp

englefly replied:
the test result： 1419 ms -> 1293 ms
tested 3 times, the average BuildTableInsertTime decreased by 8.8% (=  (1419-1293) / 1419 )

SleepyBear96 committed:
[fix](broker load) sync the workflow of BrokerScanner to other Scanner to avoid oom (#9173)

Gabriel39 committed:
[BUG] fix compiling bug for java udf (#9161)

dataroaring committed:
avoiding a corrupt image file when there is image.ckpt with non-zero … (#9180)

* avoiding a corrupt image file when there is image.ckpt with non-zero size

For now, saveImage writes data to image.ckpt via an append FileOutputStream,
when there is a non-zero size file named image.ckpt, a disaster would happen
due to a corrupt image file. Even worse, fe only keeps the lastest image file
and removes others.

BTW, image file should be synced to disk.

It is dangerous to only keep the latest image file, because an image file is
validated when generating the next image file. Then we keep an non validated
image file but remove validated ones. So I will issue a pr which keeps at least
2 image file.

* append other data after MetaHeader

* use channel.force instead of sync

stalary committed:
FIX: getChannel -> getChannel() (#9217)

Co-authored-by: Rongqian Li <rongqian_li@idgcapital.com>

yiguolei committed:
[Improvement] not print logs to fe.out when fe is running under daemon mode (#9195)

Co-authored-by: yiguolei <yiguolei@gmail.com>

dataroaring committed:
keep at least one validated image file (#9192)

* rename ImageSeq to LatestImageSeq in Storage

* keep at least one validated image file

Henry2SS committed:
[Enhancement] (image) check image validity as soon as generated (#9011)

* load newly generated image file as soon as generated to check if it is valid.

* delete the latest invalid image file

* fix

* fix

* get filePath from saveImage() to ensure deleting the correct file while exception happens

* fix

Co-authored-by: wuhangze <wuhangze@jd.com>

dataroaring committed:
[github] enable requested status check before merging pull requests (#9222)

zhangstar333 committed:
[fix](function) fix lag/lead function return invalid data (#9076)

cambyzju committed:
[feature-wip](array-type) ArrayFileColumnIterator bug fix (#9114)

spaces-X committed:
[fix](spark load) fix `getHashValue` of string type is always zero in spark load. (#9136)

Buffer flip is used incorrectly.
When the hash key is string type, the hash value is always zero.
The reason is that the buffer of string type is obtained by wrap, which is not needed to flip.
If we do so, the buffer limit for read will be zero.

morningman committed:
[fix](hierarchical-storage) Fix bug that storage medium property change back to SSD (#9158)

1. fix bug described in #9159
2. fix a `fill_tuple` bug introduced from #9173

Userwhite committed:
[fix] fix sequence bug in non-vec mode (#9184)

BiteTheDDDDt committed:
[fix](script) meet error on start_fe.sh(#9187)

start_fe.sh: line 174: [: -eq: unary operator expected

caoliang-web committed:
[refactor](doc)Cluster upgrade adds metadata backup (#9189)

BiteTheDDDDt committed:
[fix](Lateral-View)(Vectorized) core dump on lateral-view with nullable column (#9191)

Toms1999 committed:
[typo](annotation): fix typo in ldap.conf (#9200)

SleepyBear96 committed:
[fix](OlapScanner)fix bitmap or hll's OOM when loading too many unqualified data (#9205)

zhannngchen committed:
update comments in run-be-ut.sh (#9092)

Humbedooh committed:
[infra] Adjust .asf.yaml spacing to make it parse properly

mklzl committed:
Modify incorrect comments in ShowExecutor (#9232)

Fixed some incorrect comments in ShowExecutor

yangzhg committed:
remove gensrc/proto/palo_internal_service.proto, this removed in #6341 and add back in #6329 by mistake (#9233)

yangzhg committed:
[feature] add `SHOW TABLET STORAGE FORMAT` stmt (#9037)

use this stmt to show tablets storage format in be, if verbose is set,
    will show detail message of tablet storage format.
    e.g.
    ```
    MySQL [(none)]> admin show tablet storage format;
    +-----------+---------+---------+
    | BackendId | V1Count | V2Count |
    +-----------+---------+---------+
    | 10002     | 0       | 2867    |
    +-----------+---------+---------+
    1 row in set (0.003 sec)
    MySQL [test_query_qa]> admin show tablet storage format verbose;
    +-----------+----------+---------------+
    | BackendId | TabletId | StorageFormat |
    +-----------+----------+---------------+
    | 10002     | 39227    | V2            |
    | 10002     | 39221    | V2            |
    | 10002     | 39215    | V2            |
    | 10002     | 39199    | V2            |
    +-----------+----------+---------------+
    4 rows in set (0.034 sec)
    ```
    add storage format infomation to show full table statment.
    ```
    MySQL [test_query_qa]> show full tables;
    +-------------------------+------------+---------------+
    | Tables_in_test_query_qa | Table_type | StorageFormat |
    +-------------------------+------------+---------------+
    | bigtable                | BASE TABLE | V2            |
    | test_dup                | BASE TABLE | V2            |
    | test                    | BASE TABLE | V2            |
    | baseall                 | BASE TABLE | V2            |
    | test_string             | BASE TABLE | V2            |
    +-------------------------+------------+---------------+
    5 rows in set (0.002 sec)
    ```

morningman committed:
[test] reset default port in regression test conf (#9246)

Co-authored-by: morningman <chenmingyu@baidu.com>

GoGoWen committed:
remove some unused code (#9240)

weizhengte committed:
[feature-wip](statistics) step2: schedule the statistics job and generate executable tasks (#8859)

This pull request includes some implementations of the statistics(https://github.com/apache/incubator-doris/issues/6370), it will not affect any existing code and users will not be able to create statistics job.

After receiving the statistics collection statement, it generates a job. Here it implements the division of statistics collection jobs according to the following statistics categories:
table:
- `row_count`: table row count are critical in estimating cardinality and memory usage of scan nodes.
- `data_size`: table size, not applicable to CBO, mainly used to monitor and manage table size.
column:
- `num_distinct_value`: used to determine the selectivity of an equivalent expression.
- `min`: The minimum value.
- `max`: The maximum value.
- `num_nulls`: number of nulls.
- `avg_col_len`: the average length of a column, in bytes, is used for memory and network IO evaluation.
- `max_col_len`: the Max length of the column, in bytes, is used for memory and network IO evaluation.

After the job is divided, statistics tasks will be obtained.

xinyiZzz committed:
[feature-wip] (memory tracker) (step5) Fix track bthread, fix track vectorized query (#9145)

1. fix track bthread
- Bthread, a high performance M:N thread library used by brpc. In Doris, a brpc server response runs on one bthread, possibly on multiple pthreads. Currently, MemTracker consumption relies on pthread local variables (TLS).
- This caused pthread TLS MemTracker confusion when switching pthread TLS MemTracker in brpc server response. So replacing pthread TLS with bthread TLS in the brpc server response saves the MemTracker.
Ref: https://github.com/apache/incubator-brpc/blob/731730da85f6af5c25012b4c83ab5bb371320cf8/docs/en/server.md#bthread-local

2. fix track vectorized query
- Added track mmap. Currently, mmap allocates memory in many places of the vectorized execution engine.
- Refactored ThreadContext to avoid dependency conflicts and make it easier to debug.
- Fix some bugs.

qzsee committed:
[fix](planner) fix non-equal out join is not supported (#9156)

xy720 committed:
[fix](routine-load) Fix bug that new coming routine load tasks are rejected all the time and report TOO_MANY_TASK error (#9164)

```
CREATE ROUTINE LOAD iaas.dws_nat ON dws_nat
WITH APPEND PROPERTIES (
"desired_concurrent_number"="2",
"max_batch_interval" = "20",
"max_batch_rows" = "400000",
"max_batch_size" = "314572800",
"format" = "json",
"max_error_number" = "0"
)
FROM KAFKA (
"kafka_broker_list" = "xxxx:xxxx",
"kafka_topic" = "nat_nsq",
"property.kafka_default_offsets" = "2022-04-19 13:20:00"
);
```

In the create statement example below, you can see
The user didn't specify the custom partitions.
So that 1. Fe will get all kafka partitions from server in routine load's scheduler.
The user set the default offset by datetime.
So that 2. Fe will get kafka offset by time from server in routine load's scheduler.

When 1 is success, meanwhile 2 is failed, the progress of this routine load may not contains any partitions and offsets.
Nevertheless, since newCurrentKafkaPartition which is get by kafka server may be always equal to currentKafkaPartitions, 
the wrong progress will never be updated.

Gabriel39 committed:
[Improvement](docs) Update EN doc (#9228)

morningman committed:
[fix](checkpoint) fix checkpoint failure when reloading new image (#9262)

Introduced from #9011

yangzhg committed:
[chore] (workflow)add sonarcloud workflow to check code quality and security (#9252)

yiguolei committed:
[feature-wip](global-dict) global dict thrift definition (#9243)

Co-authored-by: yiguolei <yiguolei@gmail.com>

EmmyMiao87 committed:
(Refactor)[Planner] Remove merge node (#9251)

hf200012 committed:
[refactor][doc]The new version of the document is online (#9272)

replace the `docs/` with `new-docs/`

wangbo committed:
[fix](storage)bloom filter support ColumnDict (#9167)

bloom filter support ColumnDict(#9167)

morningman committed:
[fix] fix docs build bug  (#9293)

After this PR #9272, the `docs/build_help_zip.sh` will run failed.
This PR fix this issue.
But the help module still has some parse problem, I will fix it in next PR.

This CL mainly changes:
1. fix `docs/build_help_zip.sh` error
2. remove `sql-reference-v2` to `sql-reference`
3. modify build extension github action to run `docs/build_help_zip.sh`

HappenLee committed:
[Vectorized](stream-load-vec) Support stream load in vectorized engine (#8709) (#9280)

Implement vectorized stream load.
Added fe configuration option `enable_vectorized_load` to enable vectorized stream load.

    Co-authored-by: tengjp@outlook.com
    Co-authored-by: mrhhsg@gmail.com
    Co-authored-by: minghong.zhou@163.com
    Co-authored-by: HappenLee <happenlee@hotmail.com>
    Co-authored-by: zhoubintao <35688959+zbtzbtzbt@users.noreply.github.com>

xy720 committed:
[refactor][routineload] Remove unused client object from routine load (#9223)

zhengshiJ committed:
[feature](statistics) Statistics derivation.Step 1:ScanNode implement… (#8947)

* [feature](statistics) Statistics derivation.Step 1:ScanNode implementation

Co-authored-by: jianghaochen <jianghaochen@meituan.com>

wangbo committed:
[test](storage)duplicate model storage layer regression-test (#9285)

* duplicate model storage layer regression-test

* use insert values for batch

Co-authored-by: Wang Bo <wangbo36@meituan.com>

morningman committed:
[fix](materialized-view) fix bug that can not create mv for list partitioned table (#9281)

zenoyang committed:
[fix](profile) Short-circuit and del predicate filter rows are not counted on vectorized exec (#9268)

EmmyMiao87 committed:
(Refactor)[Statistics] Fix lock risks in Statistics Job (#9256)

* (Refactor)[Statistics] Fix lock risks in Statistics Job

1. Remove lock nesting between job and task
2. Solve the deadlock problem during job update
3. Avoid printing the log while holding the lock

* Add log

long2ice committed:
[Features]Add dbt doris adapter (#9299)

* Add dbt doris adapter

* Add licence header to each file

* Fix licence header

morningman committed:
[fix](help-doc) fix format of all sql-manual doc (#9306)

zenoyang committed:
[fix](vectorized) Query get wrong result when ColumnDict concurrent predicate eval (#9270)

jacktengg committed:
[Enhancement][Vectorized] Improve hash table build efficiency (#9250)

1. MAP_POPULATE is missing for mmap in Allocator, because macro OS_LINUX is not defined in allocator.h;
2. MAP_POPULATE has no effect for mremap as for mmap, zero-fill enlarged memory range explicitly to pre-fault the pages

chenlinzhong committed:
[style] clang-format all c++ code (#9305)

- sh build-support/clang-format.sh  to  clang-format all c++ code

morningman committed:
[fix](ut) Fix MarkDownParserTest (#9332)

pingchunzhang committed:
[github-action] Update regression required check in .asf.yaml (#9333)

Change `P0 regression (Doris P0 regression)` to `P0 regression (Doris P0 Regression)`

hf200012 committed:
[fix][doc]Modify readme link (#9334)

weizhengte committed:
[feature-wip](statistics) step3: schedule the statistics tasks and update relevant info (#8860)

This pull request includes some implementations of the statistics(https://github.com/apache/incubator-doris/issues/6370), it will not affect any existing code and users will not be able to create statistics job.

After receiving the statistics statement and dividing the collection task, here we will start implementing the scheduling statistics task and updating the job information. Mainly include the following:
- Create a thread pool to schedule a certain number of tasks, and the number of concurrency is related to the configuration `cbo_concurrency_statistics_task_num`.
- After the task is completed, update the information of of the statistics Job.

morrySnow committed:
[FE Code Style][step 0]add github action to check incremental code in pr (#9328)

1. add rules to checkstyle
2. add github action to check incremental code in pr

dataroaring committed:
fix complication guide on ubuntu (#9337)

stalary committed:
[doc] add error doc for action_table.dat (#9263)

JNSimba committed:
[doc] fix new doc v2 bug (#9309)

change sql-references-v2 to sql-references

JNSimba committed:
[doc] Add Flink Connector Sink Bitmap Type Doc (#9316)

JNSimba committed:
[doc] update doris manager doc (#9319)

morningman committed:
[fix](truncate) fix bug that truncate partition throw NPE (#9339)

1. partition name is case insensitive.
2. add a simple help-resource.zip to help pass the FE ut.

stalary committed:
[doc] fix EN doc edit link 404 (#9182)

BiteTheDDDDt committed:
[fix](Lateral-View) fix outer combinator not work on non-vectorized (#9212)

BiteTheDDDDt committed:
[Bug] [Build] fix clang build fail (#9323)

* fix clang compile fail

Gabriel39 committed:
[fix](compaction) fix bug for vectorized compaction (#9344)

1. add a BE config to switch vectorized compaction
2. Fix vectorized compaction bug that row statistic is not right.

morningman committed:
[fix](catalog) fix bug that replica missing version cause query -214 error (#9266)

1. Fix bug described in #9267 
    When report missing version replica, set last failed version to (replica version + 1)
2. Skip non-exist partition when handling transactions.

emerkfu committed:
[doc] Update sequence-column-manual.md (#9321)

morningman committed:
[fix](alter-job) Missing alter job when doing checkpoint image (#9329)

This bug is introduced from #8030

dataroaring committed:
[fix](function) handle merge in window_funnel_init and add test (#9338)

Gabriel39 committed:
[doc] Remove deprecated BE config and add new config (#9347)

stalary committed:
[doc] Implement java-format-code.md for IDEA (#9349)

dataalive committed:
[typo] fix getting started doc (#9352)

yiguolei committed:
[refactor][storage format] Forbidden rowset v1 (#9248)

- Force change the existing olaptable's storage format from V1 to V2
- Forbidden to create new olap table with storage format == v1 OR do schema change that want to create new v1 format

xleoken committed:
[FE Code Style][sub] Adjust some check rules (#9345)

Adjust `RedundantImport`,`UnusedImports`,`EmptyStatement`,`NewlineAtEndOfFile`,`UpperEll`, `AvoidStarImport`, `MissingOverride` rules.

hf200012 committed:
[fix][doc]add design doc (#9324)

add design doc

HappenLee committed:
[fix](vectorized) Fix bug of outer join with delete column (#9315)

924060929 committed:
[improvement](regression-test) support exclude suite/group/directory (#9096)

regression testing framework support skip some suite/group/directory

dataroaring committed:
[improvement] set name for scanner threads and fix compile error in clang (#9336)

HappenLee committed:
[fix](memtracker) DCHECK failed in vetorized exec engine fold constant execute (#9354)

liuzhuang2017 committed:
[docs] Fix some typos in documentation. (#9356)

hf200012 committed:
[improvement](load) Add http header size parameter (#9357)

Add the http header size parameter to avoid failure due to too many fields when users import using stream load.
The normal default is 8192, and 10K is given here.

luzhijing committed:
[doc] update member.md (#9361)

yiguolei committed:
[fixbug][metadata] catalog could not load from image (#9364)

* [fixbug][catalog] catalog could not load from image

* fix ut failed

Co-authored-by: yiguolei <yiguolei@gmail.com>

morningman committed:
[fix](backup) Remove colocate_with property when backing up a table (#9142)

We currently not support backup table with colocation property.
So that we have to remove colocate_with property from a table when backing up.

yangzhg committed:
[improvement](odbc) support more ODBC Connection Parameters for odbc external table (#9198)

user can add more supported ODBC Connection Parameters to resource PROPERTIES or external table PROPERTIES

924060929 committed:
[improvement](regression-test) Support suite plugin to add third-part… (#9294)

Support register suite plugin to add third-party function.

See 
1. register in: ${DORIS_HOME}/regression-test/plugins/plugin_example.groovy
2. usage: ${DORIS_HOME}/regression-test/suites/demo/test_plugin.groovy
3. doc: ${DORIS_HOME}/docs/zh-CN/developer-guide/regression-testing.md

morningman committed:
[doc][community] add more collaborator and modify documents (#9320)

1. Add collaborator morrySnow
2. Add document: How To Be A Committer
3. Add guidance to install clang-format 11

dataalive committed:
[doc]fix geo funtion st_distance_sphere (#9341)

add ST_Distance_Sphere function parameter values range restrictions

morningman committed:
[fix](ut) fix a potential memory leak in BE ut (#9362)

qimumu9406 committed:
[doc] The description of the metadata design document is not clear (#9367)

Jibing-Li committed:
[improvement](hive) Support hive with HA HDFS. Pass ha configuration through hive create table properties. (#9151)

Doris couldn't resolve the defaultFS of HDFS with HA configuration, so it could query hive table on HA HDFS.
This is because there's no way to send the HA configs to hive external table.

Describe the overview of changes.
Pass the ha configs to hive external table through create table properties.

Usage:
Example of creating hive table with ha configuration properties:

CREATE TABLE region (
r_regionkey integer NOT NULL,
r_name char(25) NOT NULL,
r_comment varchar(152)
) engine=hive properties
("database"="default",
"table"="region",
"hive.metastore.uris"="thrift://172.21.16.11:7004",
"dfs.nameservices"="hacluster",
"dfs.ha.namenodes.hacluster"="3,4",
"dfs.namenode.rpc-address.hacluster.3"="192.168.0.93:8020",
"dfs.namenode.rpc-address.hacluster.4"="172.21.16.11:8020", "dfs.client.failover.proxy.provider.hacluster"="org.apache.hadoop.hdfs.server.namenode.ha.ConfiguredFailoverProxyProvider");

yiguolei committed:
remove file result writer from result sink (#9378)

xinyiZzz committed:
[fix][compaction]  Rowset::end_version null pointer(#9379)

caoliang-web committed:
[doc]Add binlog to load and import the doris table and the original table … (#9317)

long2ice committed:
[Feature][dbt] add partition_type support (#9389)

HappenLee committed:
[Bug][stream-vec-load] Null data load do not skip the same place data (#9360)

Co-authored-by: lihaopeng <lihaopeng@baidu.com>

morrySnow committed:
[WIP-feature](Optimizer) Nereids code base (#9392)

Nereids(new optimizer) code base

Nereids is new query planner for Doris. It include three main parts: parser, analyzer and optimizer.
The parser, generated by ANTLR4, transforms SQL into a logical plan with a tree structure. Analysis and optimization are performed on the logical plan of the tree structure. Each transformation is defined as a rule. The rule is applied to the logical plan using pattern matching. The implementation of the optimizer follows the approach in the Cascades paper.

lonre committed:
[feature][datax]doriswriter support timeZone (#9327)

wangbo committed:
[refactor]refactor lazy materialized (#8834)

[refactor]refactor lazy materialized (#8834)

Gabriel39 committed:
[fixbug][compaction] update OLAP_ERR_CUMULATIVE_NO_SUITABLE_VERSION (#9410)

nvguiyachuang committed:
[docs]docs ddl modify (#9415)

SaintBacchus committed:
[doc]Fix some path error in regression-testing doc (#9413)

qzsee committed:
[performance][query]improve the performance of DISTINCT aggregation by using flat hash set replace unordered set (#9401)

Co-authored-by: shizhiqiang03 <shizhiqiang03@meituan.com>

hf200012 committed:
[fix][doc]Chinese and English homepage link error (#9397)

Chinese and English homepage link error

morningman committed:
[doc] Add window functions sql help doc (#9393)

hf200012 committed:
[fix][doc]style issues (#9386)

morningman committed:
[doc] add github checks doc (#9387)

Gabriel39 committed:
[Improvement] [compaction]Enable vectorized compaction by default (#9383)

dataalive committed:
[doc]add error reason for storage medium (#9390)

stalary committed:
[Feature] CTAS support insert data (#9271)

yangzhg committed:
[enhancement](load) optimize load string data and dict page write (#9123)

* [enhancement](load) optimize load string data and dict page write

BiteTheDDDDt committed:
[Enhancement] [Vectorized] Refactor and optimize BinaryOperation (#9087)

Gabriel39 committed:
[BUG][Vectorized] fix `replace_if_not_null` in vectorized compaction (#9376)

hf200012 committed:
[doc]Add hll usage documentation (#9398)

Add hll usage documentation

chenlinzhong committed:
[Bug] (fix) DeltaWriter::mem_consumption() coredump (#9245)

hddong committed:
[DOC]Improve fe idea dev (#9431)

dataalive committed:
[doc]fix typo in en faq and format pattern (#9423)

stalary committed:
[Bug][CTAS] create table by partition list (#9412)



Co-authored-by: Rongqian Li <rongqian_li@idgcapital.com>

xleoken committed:
[FE Code Style] Fix wrong checkstyle action configuration (#9370)

caoliang-web committed:
[doc]Add create java udf document (#9430)

* Add create java udf document

* Add create java udf document

hf200012 committed:
[fix]Stream load 307 temporary redirection authentication information is lost (#9363)

zhannngchen committed:
[refactor] some code cleanup for min/max function. (#8874)

zhengshiJ committed:
[fix](rewrite) The where condition cannot be pushed down because there is no derivation (#8980)

Fix a bug.
The where condition cannot be pushed down because there is no derivation

eg:
select * from tb1 left join tb2 on tb1.id = tb2.id where tb2.id = 1;

The correct case is that the condition of "=1" needs to be deduced to tb1.id,
but the current implementation does not do the deduction

zhangstar333 committed:
[fix][feature](Function) fix return type && support hll_union_agg/group_concat agg to window function (#9119)

wangshuo128 committed:
[refactor](test) Refactor FE unit test framework that starts a FE server. (#9388)

Currently, we use `UtFrameUtils` to start a FE server in the FE unit test. 
Each test class has to do some initialization and clean up stuff with the JUnit4
`@BeforeClass` and `@AfterClass` annotation. It's redundant and boring.
Besides, almost all the APIs in `UtFrameUtils` has a `ConnectContext` parameter, which is not easy to use.

This PR proposes to use an inherit-manner, i.e., wrap all the common logic in base class `TestWithFeService`,
leveraging the 
JUnit5 `@BeforeAll` and `@AfterAll` annotation to narrow down the setup and cleanup lifecycle to each test class instance.
At the same time, the derived concrete test class could directly use utility methods inherited from the base class,
without calling a util class and passing a `ConnectContext` argument.

`UtFrameUtils` and `DorisAssert`  are marked as deprecated. We could remove these two classes
if this refactor works well for a time.

cxzl25 committed:
improve doc spark3.2.0 (#9445)

morrySnow committed:
[style]add IntelliJ code style xml (#9438)

pengxiangyu committed:
[Bug] Missing error tablet list when close_wait return error (#9418)

924060929 committed:
[Enhancement](Optimizer) Optimize nereids tree node structure (#9446)

This pr optimize nereids tree node structure for generic parameter and Nary abstract tree node.
It can facilitate the use of pattern match framework.

Henry2SS committed:
[feature] (sql-digest) support sql digest (#8919)

caiconghui committed:
[fix](schema_hash) Fix bug that introduced by removing schema_hash (#9449)

JNSimba committed:
[Doc] fix doc link suffix .html to .md (#9442)

* fix doc link suffix html to md

sparklezzz committed:
Update data-partition.md (#9450)

Update data-partition.md

emerkfu committed:
[doc] Update DECIMAL.md (#9451)

* Update DECIMAL.md

BePPPower committed:
[doc] update data-model.md and data-partition.md (#9448)

update data-model.md and data-partition.md

yangzhg committed:
[fix](s3) fix s3 Temp file may write failed because of has no space on disk (#9421)

cambyzju committed:
[fix] UT MathFunctionTest.round_test fix (#9447)

Function round support two format round(double) and round(double, int), the argument is variadic.
But FunctionBinaryArithmetic not support variadic argument now, make get_function for round(double, int) failed.

reproduce steps:
1. set enable_vectorized_engine=true;
2. try to call round(double, int);
```
> select round(10.12345,2);
ERROR 1105 (HY000): errCode = 2, detailMessage = Function round is not implemented
```

xinyiZzz committed:
[fix](ut) fix DeltaWriter::close_wait parameter mismatch in delta_writer_test (#9457)

xleoken committed:
[code style](fe) Include test sources (#9366)

Include test sources, we also need to check them.

xiepengcheng01 committed:
[feature] (vec)  instead of converting line to src tuple for stream load in vectorized. (#9314)

Co-authored-by: xiepengcheng01 <xiepengcheng01@xafj-palo-rpm64.xafj.baidu.com>

stalary committed:
[Doc] 添加CTAS文档 (#9454)

* ADD: 添加CTAS文档

liuzhuang2017 committed:
[docs][typo] Fix some typoes in "update.md" content. (#9455)

Fix some typoes in "update.md" content

BePPPower committed:
[refactor] modify all OLAP_LOG_WARNING to LOG(WARNING) (#9473)

Co-authored-by: BePPPower <fangtiewei@selectdb.com>

xlwh committed:
[Refactor] Replace '#ifndef' with '#pragma once' (#9456)

* Replace '#ifndef' with '#pragma once'

924060929 committed:
[Enhancement](Optimizer) Nereids pattern matching base framework (#9474)

This pr provide a new pattern matching framework for Nereids optimizer.

The new pattern matching framework contains this concepts:

1. `Pattern`/`PatternDescriptor`: the tree node's multiple hierarchy shape, e.g. `logicalJoin(logicalJoin(), any()` pattern describe a plan that root is a `LogicalJoin` and the left child is `LogicalJoin` too.
2. `MatchedAction`: a callback function when the pattern matched, usually you can create new plan to replace the origin matched plan.
3. `MatchingContext`: the param pass through MatchedAction, contains the matched plan root and the PlannerContext.
4. `PatternMatcher`: contains PatternDescriptor and MatchedAction
5. `Rule`: a rewrite rule contains RuleType, PatternPromise, Pattern and transform function(equals to MatchedAction)
6. `RuleFactory`: the factory can help us build Rules easily. RuleFactory extends Patterns interface, and have some predefined pattern descriptors.

for example, Join commutative:
```java
public class JoinCommutative extends OneExplorationRuleFactory {
    @Override
    public Rule<Plan> build() {
        return innerLogicalJoin().thenApply(ctx -> {
            return new LogicalJoin(
                JoinType.INNER_JOIN,
                ctx.root.getOnClause(),
                ctx.root.right(),
                ctx.root.left()
            );
        }).toRule(RuleType.LOGICAL_JOIN_COMMUTATIVE);
    }
}
```

the code above show the three step to create a Rule
1. 'innerLogicalJoin()' declare pattern  is an inner logical join. 'innerLogicalJoin' is a predefined pattern.
2. invoke 'thenApply()' function to combine a MatchedAction, return a new LogicalJoin with exchange children.
3. invoke 'toRule()' function to convert to Rule

You can think the Rule contains three parts: 
1. Pattern
2. transform function / MatchedAction
3. RuleType and RulePromise

So
1. `innerLogicalJoin()` create a `PatternDescriptor`, which contains a `Pattern`
2. `PatternDescriptor.then()` convert `PatternDescriptor` to `PatternMatcher,` witch contains Pattern and MatchedAction
3. `PatternMatcher.toRule()` convert `PatternMatcher` to a Rule

This three step inspired by the currying in function programing.

It should be noted, #9446 provide a generic type for TreeNode's children, so we can infer multiple hierarchy type in this pattern matching framework, so you can get the really tree node type without unsafely cast. like this:
```java
logicalJoin(logicalJoin(), any()).then(j -> {
     // j can be inferred type to LogicalJoin<LogicalJoin<Plan, Plan>, Plan>
     // so j.left() can be inferred type to LogicalJoin<Plan, Plan>,
     // so you don't need to cast j.left() from 'Plan' to 'LogicalJoin'
     var node = j.left().left();
})
```

xinyiZzz committed:
[feature-wip] (memory tracker) (step6, End) Fix some details  (#9301)

1. Fix LoadTask, ChunkAllocator, TabletMeta, Brpc, the accuracy of memory track.
2. Modified some MemTracker names, deleted some unnecessary trackers, and improved readability.
3. More powerful MemTracker debugging capabilities.
4. Avoid creating TabletColumn temporary objects and improve BE startup time by 8%.
5. Fix some other details.

BePPPower committed:
[refactor] delete OLAP_LOG_WARNING related macro definition (#9484)

Co-authored-by: BePPPower <fangtiewei@selectdb.com>

adonis0147 committed:
[refactor][style] Use clang-format to sort includes (#9483)

stalary committed:
[feature] show create materialized view (#9391)

nextdreamblue committed:
[feature](mysql-table) support utf8mb4 for mysql external table (#9402)

This patch supports utf8mb4 for mysql external table.

if someone needs a mysql external table with utf8mb4 charset, but only support charset utf8 right now.

When create mysql external table, it can add an optional propertiy "charset" which can set character fom mysql connection, 
default value is "utf8". You can set "utf8mb4" instead of "utf8" when you need.

Gabriel39 committed:
[regression] add regression test for compaction (#9437)

Trigger compaction via REST API in this case.

morningman committed:
[refactor](backend) Refactor the logic of selecting Backend in FE. (#9478)

There are many places in FE where a group of BE nodes needs to be selected according to certain requirements. for example:
1. When creating replicas for a tablet.
2. When selecting a BE to execute Insert.
3. When Stream Load forwards http requests to BE nodes.

These operations all have the same logic. So this CL mainly changes:
1. Create a new `BeSelectionPolicy` class to describe the set of conditions for selecting BE.
2. The logic of selecting BE nodes in `SystemInfoService` has been refactored, and the following two methods are used uniformly:
    1. `selectBackendIdsByPolicy`: Select the required number of BE nodes according to the `BeSelectionPolicy`.
    2. `selectBackendIdsForReplicaCreation`: Select the BE node for the replica creation operation.

Note that there are some changes here:
For the replica creation operation, the round-robin method was used to select BE nodes before,
but now it is changed to `random` selection for the following reasons:
1. Although the previous logic is round-robin, it is actually random.
2. The final diff of the random algorithm will not be greater than 5%, so it can be considered that the random algorithm
     can distribute the data evenly.

hf200012 committed:
[fix](http) Hardening Recommendations Disable TRACE/TRAC methods (#9479)

jackwener committed:
[refactor](Nereids): cascades refactor (#9470)

Describe the overview of changes.

- rename GroupExpression
- use `HashSet<GroupExpression> groupExpressions` in `memo`
- add label of `Nereids` for CI
- remove `GroupExpr` from Plan

morrySnow committed:
[doc] update fe checkstyle doc (#9373)

yixiutt committed:
[bugfix](vtablet_sink) fix max_pending_bytes for vtablet_sink (#9462)



Co-authored-by: yixiutt <yixiu@selectdb.com>

pengxiangyu committed:
[fixbug]fix bug for OLAP_SUCCESS with Status (#9427)

stalary committed:
[feature] support row policy filter (#9206)

morrySnow committed:
[chore](fe code style)add suppressions to fe check style (#9429)

Current fe check style check all files. But some rules should be only applied on production files.
Add suppressions to suppress some rules on test files.

deardeng committed:
[fix](broker-load) can't load parquet file with column name case sensitive with Doris column (#9358)

hf200012 committed:
[fix](binlog-load) binlog load fails because txn exceeds the default value (#9471)

binlog load Because txn exceeds the default value, resume is a failure,
and a friendly prompt message is given to the user, instead of prompting success now,
it still fails after a while, and the user will feel inexplicable
Issue Number: close #9468

xlwh committed:
[refactor]Cleanup unused empty files (#9497)

platoneko committed:
[refactor] Check status precise_code instead of construct OLAPInternalError (#9514)

* check status precise_code instead of construct OLAPInternalError
* move is_io_error to Status

wangbo committed:
[fix](storage) fix core for string predicate in storage layer (#9500)



Co-authored-by: Wang Bo <wangbo36@meituan.com>

MorningLight5 committed:
[Bug] (load) Broker load kerberos auth fail (#9494)

smallhibiscus committed:
Incorrect sequence numbers in revision documents. (#9496)

Co-authored-by: smallhibiscus <844981280>

carlvinhust2012 committed:
[regression test]add the regression test for json load (#9517)

Co-authored-by: hucheng01 <hucheng01@baidu.com>

morrySnow committed:
[style](java) format fe code with some check rules (#9460)

Issue Number: close #9403 

set below rules' severity to error and format code according check info.
a. Merge conflicts unresolved
b. Avoid using corresponding octal or Unicode escape
c. Avoid Escaped Unicode Characters
d. No Line Wrap
e. Package Name
f. Type Name
g. Annotation Location
h. Interface Type Parameter
i. CatchParameterName
j. Pattern Variable Name
k. Record Component Name
l. Record Type Parameter Name
m. Method Type Parameter Name
n. Redundant Import
o. Custom Import Order
p. Unused Imports
q. Avoid Star Import
r. tab character in file
s. Newline At End Of File
t. Trailing whitespace found

yixiutt committed:
[bugfix](load) fix coredump in ordinal index flush (#9518)

commit #9123 introduce the bug. bitshuffle page return error when
page is full, so scalar column write cannot switch to next page, which make
ordinal index is null when flush.

All page builder should return ok when page full, and column writer procedure
shoud be append_data, check is_page_full, switch to next page

Co-authored-by: yixiutt <yixiu@selectdb.com>

yinzhijian committed:
[fix][vectorized-storage] did not check column writer's write status

hello-stephen committed:
Clean the version.sh file before build, otherwise the version information in the binary package produced by this compilation is still the commit id of the last time. (#9534)

Co-authored-by: stephen <hello-stephen@qq.com>

FreeOnePlus committed:
[doc]Add ARM architecture compilation tutorial content (#9535)

Co-authored-by: manyi <fop@freeoneplus.com>

cambyzju committed:
[feature-wip](array-type) array_contains support more nested data types (#9170)

Co-authored-by: cambyzju <zhuxiaoli01@baidu.com>

Gabriel39 committed:
[Improvement] remove unnecessary memcpy in OlapBlockDataConvertor (#9491)

* [Improvement] remove unnecessary memcpy in OlapBlockDataConvertor

LOVEGISER committed:
[doc] [Improved] The flink connector documentation is perfect (#9528)


Co-authored-by: 王磊 <lei.wang@unidt.com>

carlvinhust2012 committed:
[feature] add vectorized vjson_scanner (#9311)

This pr is used to add the vectorized vjson_scanner, which can support vectorized json import in stream load flow.

zhangstar333 committed:
[fix](Function) fix case when function return null with abs function (#9493)

EmmyMiao87 committed:
[fix](lateral-view) Error view includes lateral view (#9530)

Fixed #9529

When the lateral view based on a inline view which belongs to a view,
Doris could not resolve the column of lateral view in query.
When a query uses a view, it mainly refers to the string representation of the view.
That is, if the view's string representation is wrong, the view is wrong.
The string representation of the inline view lacks the handling of the lateral view.
This leads to query errors when using such views.
This PR mainly fixes the string representation of inline views.

yiguolei committed:
[refactor](es) Clean es tcp scannode and related thrift definitions (#9553)

PaloExternalSourcesService is designed for es_scan_node using tcp protocol.
But es tcp protocol need deploy a tcp jar into es code. Both es version and lucene version are upgraded,
and the tcp jar is not maintained any more.

So that I remove all the related code and thrift definitions.

jacktengg committed:
[bugfix](vectorized) vectorized write: invalid memory access caused by podarray resize (#9556)

stalary committed:
ADD: 补充idea开发文档，添加help-resource.zip的生成步骤 (#9561)

dataalive committed:
[doc]fix doc typo in data-model and date data type (#9571)

hf200012 committed:
[Doc]Add show tables help documentation (#9568)

xiaokang committed:
[enhancement][betarowset]optimize lz4 compress and decompress speed by reusing context (#9566)

zhangstar333 committed:
[fix](function) fix last_value get wrong result when have order by clause (#9247)

EmmyMiao87 committed:
[Feature](Nereids) Data structure of comparison predicate (#9506)

1. The data structure of the comparison expression
2. Refactored the inheritance and implementation relationship of tree node

```
        +-- ---- ---- ---+- ---- ---- ---- ---+- ---- ----- ---- ----TreeNode-----------------+
        |                |                    |                                               |
                                                                                              |
        |                |                    |                                               |
                                                                                              v
        v                v                    v                                           Abstract Tree Node
    Leaf Node        Unary Node          Binary Node                              +--------          ---------+
        |                |                    |                                   |        (children)         |
                                                                                  |                           |
        v                v                    v                                   v                           v
Leaf Expression   Unary Expression      Binary Expression              +------Expression----+           Plan Node
        |                |                    |                        |                    |
                                                                       |                    |
        |                |                    |                        v                    v
        |                |                    +- ---- ---- -----> Comparison Predicate     Named Expr
                                                                                       +----   -------+
        |                |                                                             v              v
        |                +- -- --- --- --- --- --- --- --- --- --- --- --- --- ---> Alias Expr      Slot
                                                                                                      ^
        |                                                                                             |
        |                                                                                             |
        +---- --- ---- ------ ---- ------- ------ ------- --- ------ ------ ----- ---- ----- ----- ---+
```

morrySnow committed:
[fix](planner)VecNotImplException thrown when query need rewrite and some slot cannot changed to nullable (#9589)

adonis0147 committed:
[chore] Fix compilation errors reported by clang (#9584)

zy-kkk committed:
[docs]Modifide flink-doris-connector.md (#9595)

yinzhijian committed:
[feature-wip](parquet-vec) Support parquet scanner in vectorized engine (#9433)

dujl committed:
[feature-wip](hudi) Step1: Support create hudi external table (#9559)

support create hudi table
support show create table for hudi table

### Design
1. create hudi table without schema(recommanded)
```sql
    CREATE [EXTERNAL] TABLE table_name
    ENGINE = HUDI
    [COMMENT "comment"]
    PROPERTIES (
    "hudi.database" = "hudi_db_in_hive_metastore",
    "hudi.table" = "hudi_table_in_hive_metastore",
    "hudi.hive.metastore.uris" = "thrift://127.0.0.1:9083"
    );
```

2. create hudi table with schema
```sql
    CREATE [EXTERNAL] TABLE table_name
    [(column_definition1[, column_definition2, ...])]
    ENGINE = HUDI
    [COMMENT "comment"]
    PROPERTIES (
    "hudi.database" = "hudi_db_in_hive_metastore",
    "hudi.table" = "hudi_table_in_hive_metastore",
    "hudi.hive.metastore.uris" = "thrift://127.0.0.1:9083"
    );
```
When create hudi table with schema, the columns must exist in corresponding table in hive metastore.

liaoxin01 committed:
[fix](storage-vectorized) fix VMergeIterator core dump (#9564)

It could be re appeared on rowset with many segment, it means segment overlap. Maybe could not reappear it easily.

HappenLee committed:
[Bug][Vectorized] Fix BE crash with delete condition and enable_storage_vectorization (#9547)

Co-authored-by: lihaopeng <lihaopeng@baidu.com>

HappenLee committed:
[Bug][Vectorized] Fix insert bimmap column with nullable column (#9408)

Co-authored-by: lihaopeng <lihaopeng@baidu.com>

hf200012 committed:
[doc]add largeint doc (#9609)

add largeint doc

zy-kkk committed:
[doc]modified the spark-load doc (#9605)

adonis0147 committed:
[code format]Upgrade clang-format in BE Code Formatter from 8 to 13 (#9602)

stalary committed:
[feature] group_concat support distinct (#9576)

pengxiangyu committed:
[feature] Add StoragePolicyResource for Remote Storage (#9554)

Add StoragePolicyResource for Remote Storage

morningman committed:
[fix] fix bug that replica can not be repaired duo to DECOMMISSION state (#9424)

Reset state of replica which state are in DECOMMISSION after finished scheduling.

morningman committed:
[config] Remove some old config and session variable (#9495)

1. Remove session variable "enable_lateral_view"
2. Remove Fe config: enable_materialized_view
3. Remove Fe config: enable_create_sync_job
4. Fe config dynamic_partition_enable is only used for disable dynamic partition scheduler.

Gabriel39 committed:
[Improvement] reduce string size in serialization (#9550)

jacktengg committed:
[Improvement][ASAN] make BE can exit normally and ASAN memory leak checking work (#9620)

BiteTheDDDDt committed:
[clang build]fix clang compile error (#9615)

carlvinhust2012 committed:
[regression test] add some case for json load regression test (#9614)

Co-authored-by: hucheng01 <hucheng01@baidu.com>

cambyzju committed:
[BUG] fix information_schema.columns results not correctly on vec engine (#9612)

* VSchemaScanNode get_next bugfix

* add regression-test case for VSchemaScanNode

Co-authored-by: cambyzju <zhuxiaoli01@baidu.com>

MorningLight5 committed:
[bug] (init) Java version check fail (#9607)

morrySnow committed:
[improment](planner) push down predicate past two phase aggregate (#9498)

Push down predicate past aggregate cannot push down predicate past 2 phase aggregate.

origin plan is like this:
```
second phase agg (conjuncts on olap scan node tuples)
|
first phase agg
|
olap scan node
```
should be optimized to
```
second phase agg
|
first phase agg
|
olap scan node (conjuncts on olap scan node tuples)
```

yixiutt committed:
[fixbug](vec-load) fix core of segment_writer while it is not thread-safe (#9569)

introduce in stream-load-vec #9280, it will cause multi-thread
operate to same segment_write cause BetaRowset enable multi-thread
of memtable flush, memtable flush call rowset_writer.add_block, it
use member variable _segment_writer to write, so it will cause
multi-thread in segment write.

Co-authored-by: yixiutt <yixiu@selectdb.com>

zenoyang committed:
[fix](storage) low_cardinality_optimize core dump when is null predicate (#9586)

Issue Number: close #9555
Make the last value of the dictionary null, when ColumnDict inserts a null value,
add the encoding corresponding to the last value of the dictionary·

Gabriel39 committed:
[regression test] Add compaction regression test case for different data models (#9660)

morrySnow committed:
[fix](planner) unnecessary cast will be added on children in CaseExpr sometimes (#9600)

unnecessary cast will be added on children in CaseExpr because use symbolized equal to compare to `Expr`'s type.
it will lead to expression compare mistake and then lead to expression substitute failed when use `ExprSubstitutionMap`

yangzhg committed:
[website] fix doris website with no link to the Privacy Policy. (#9665)

All websites must link to the Privacy Policy

zy-kkk committed:
[doc] Fixed a error in the Bitmap Index section of the document (#9679)

Gabriel39 committed:
[refactor][regressiontest] reorder license header and import statement (#9672)

EmmyMiao87 committed:
[FeConfig](Project) Project optimization is enabled by default (#9667)

JNSimba committed:
[doc]update streamload 2pc doc (#9651)

Co-authored-by: wudi <>

Gabriel39 committed:
[BUG] fix bug for vectorized compaction and some storage vectorization bug (#9610)

morrySnow committed:
[style](fe) code correct rules and name rules (#9670)

* [style](fe) code correct rules and name rules

* revert some change according to comments

adonis0147 committed:
[enhancement] Improve debugging experience. (#9677)

stalary committed:
[Feature] cancel load support state (#9537)

Yulv-git committed:
Fix some typos for docs. (#9680)

Yulv-git committed:
Fix some typos in be/. (#9681)

SaintBacchus committed:
[Bug] Fix timestamp_diff issue when timeunit is year and month (#9574)

yangzhg committed:
[fix] fix Code Quality Analysis failed (#9685)

Commit:
[improvement][performance] improve lru cache resize performance and memory usage (#9521)

BiteTheDDDDt committed:
[Bug][Vectorized] fix schema change add varchar type column default value get wrong result (#9523)

xleoken committed:
[Enhance] Add host info to heartbeat error msg (#9499)

Lchangliang committed:
[Enhancement]  improve parquet reader via arrow's prefetch and multi thread (#9472)

* add ArrowReaderProperties to parquet::arrow::FileReader

* support perfecth batch

spaces-X committed:
[fix](sparkload): fix min_value will be negative number when `maxGlobalDictValue`  exceeds integer range (#9436)

yiguolei committed:
[refactor][rowset]move rowset writer to a single place (#9368)

jackwener committed:
[feature](nereids): add join rules base code (#9598)

HappenLee committed:
[docs] Fix error command of meta tool docs (#9590)

Co-authored-by: lihaopeng <lihaopeng@baidu.com>

dataroaring committed:
[improvement](stream-load) adjust read unit of http to optimize stream load (#9154)

Jibing-Li committed:
[fix](broker-scan-node) Remove trailing spaces in broker_scanner. Make it consistent with hive and trino behavior. (#9190)

Hive and trino/presto would automatically trim the trailing spaces but Doris doesn't.
This would cause different query result with hive.

Add a new session variable "trim_tailing_spaces_for_external_table_query".
If set to true, when reading csv from broker scan node, it will trim the tailing space of the column

zhangstar333 committed:
[Vectorized][java-udf] add datetime&&largeint&&decimal type to java-udf (#9440)

HappenLee committed:
[Refactor][Bug-Fix][Load Vec] Refactor code of basescanner and vjson/vparquet/vbroker scanner (#9666)

* [Refactor][Bug-Fix][Load Vec] Refactor code of basescanner and vjson/vparquet/vbroker scanner
1. fix bug of vjson scanner not support `range_from_file_path`
2. fix bug of vjson/vbrocker scanner core dump by src/dest slot nullable is different
3. fix bug of vparquest filter_block reference of column in not 1
4. refactor code to simple all the code

It only changed vectorized load, not original row based load.

Co-authored-by: lihaopeng <lihaopeng@baidu.com>

Gabriel39 committed:
[code style] minor update for code style (#9695)

jackwener committed:
[enhancement](community): enhance java style (#9693)

Enhance java style.

Now: checkstyle about code order is in this page--Class and Interface Declarations

This pr can make idea auto rearrange code

xiepengcheng01 committed:
[Refactor] add vpre_filter_expr for vectorized to improve performance (#9508)

hf200012 committed:
[doc]Add insert best practices (#9723)

Add insert best practices

gtchaos committed:
[deps] libhdfs3 build enable kerberos support (#9524)

Currently, the libhdfs3 library integrated by doris BE does not support accessing the cluster with kerberos authentication 
enabled, and found that kerberos-related dependencies（gsasl and krb5） were not added when build libhdfs3.

so, this pr will enable kerberos support and rebuild libhdfs3 with dependencies gsasl and krb5:

- gsasl version: 1.8.0
- krb5 version: 1.19

xy720 committed:
[Refactor] simplify some code in routine load (#9532)

pengxiangyu committed:
[refactor](load) add tablet errors when close_wait return error (#9619)

dataroaring committed:
[fix] NullPredicate should implement evaluate_vec (#9689)

select column from table where column is null

zxealous committed:
[doc] Fix typos in documentation (#9692)

morningman committed:
[config](checksum) Disable consistency checker by default (#9699)

Disable by default because current checksum logic has some bugs.
And it will also bring some overhead.

Jibing-Li committed:
[doc] Add trim_tailing_spaces_for_external_table_query variable to the docs. (#9701)

zhengshiJ committed:
[improvement](planner) Backfill the original predicate pushdown code (#9703)

Due to the current architecture, predicate derivation at rewrite cannot satisfy all cases,
because rewrite is performed on first and then where, and when there are subqueries, all cases cannot be derived.
So keep the predicate pushdown method here.

eg.
select * from t1 left join t2 on t1 = t2 where t1 = 1;

InferFiltersRule can't infer t2 = 1, because this is out of specification.

The expression(t2 = 1) can actually be deduced to push it down to the scan node.

Gabriel39 committed:
[doc] update docs for FE UT (#9718)

yiguolei reviewed (dismissed):
LGTM

Commit:
clang format
